frontend/src/components/NavBar.js

```diff
@@ -107,10 +110,130 @@ class NavBar extends React.Component<{
             });
     }
 
+    search(keyword) {
+        const url = '/rest/interpret/wsearch';
+
+        $.ajax({
+            url: url,
+            dataType: 'json',
+            type: 'POST',
+            cache: false,
+            data: JSON.stringify({ keywords: [keyword] }),
+            contentType: 'application/json; charset=utf-8',
+            success: (data) => {
+                console.log(data);
+            },
+            error: (xhr, status, err) => {
+                console.error(xhr.responseText);
+            },
+        });
+    }
+
+    handleChange = (e, { value }) => this.setState({ value });
+
+    handleResultSelect = (e, { result }) => {
+        window.open('/worksheets/' + result.uuid, '_self');
+    };
+
+    initialState = { isLoading: false, results: [], value: '' };
+
+    resultRenderer = ({ title, description }) => [
+        <div key='content' className='content'>
+            {title && <div dangerouslySetInnerHTML={{ __html: title }} className='title'></div>}
+            {description && (
+                <div
+                    dangerouslySetInnerHTML={{ __html: description }}
+                    className='description'
+                ></div>
+            )}
+        </div>,
+    ];
+
+    handleSearchChange = (e, { value }) => {
+        this.setState({ isLoading: true, value });
+
+        setTimeout(() => {
+            if (this.state.value.length < 1) return this.setState(this.initialState);
+            const keywords = this.state.value.split(' ');
+            const regexKeywords = keywords.join('|');
+            const re = new RegExp(regexKeywords, 'gi');
+
+            const url = '/rest/interpret/wsearch';
+
+            $.ajax({
+                url: url,
+                dataType: 'json',
+                type: 'POST',
+                cache: false,
+                data: JSON.stringify({ keywords: keywords }),
+                contentType: 'application/json; charset=utf-8',
+                success: (data) => {
+                    // build dictionary similar to the following
+                    // {
+                    //     "name": "codalab",
+                    //     "results": [
+                    //         {
+                    //             "title": "Brakus Group",
+                    //             "description": "Cloned interactive Graphic Interface",
+                    //         },
+                    //     ]
+                    // }
+                    const filteredResults = _.reduce(
+                        data.response,
+                        (memo, item) => {
+                            item.description = item.name.replace(re, '<mark>$&</mark>');
+                            item.title = item.title.replace(re, '<mark>$&</mark>');
```

Comment:
```suggestion
                            item.title = (item.title || "").replace(re, '<mark>$&</mark>');
```

I get this error because sometimes a worksheet's title will be equal to `null`.

![image](https://user-images.githubusercontent.com/1689183/80749344-4d320b00-8af4-11ea-8b79-ceb1ee081880.png)